Source/NSDictionary+HYPNestedAttributes.m

```diff
@@ -30,14 +30,38 @@ - (NSDictionary *)nestedAttributes:(HYPNestedAttributesType)type
     NSMutableDictionary *nestedDictionaryAttributes;
 
     NSArray *sortedKeys = [[self allKeys] sortedArrayUsingSelector:@selector(compare:)];
+    NSString *currentRelation;
+    NSString *currentRelationIndex;
+    NSInteger index = 0;
+
     for (NSString *key in sortedKeys) {
         HYPParsedRelationship *parsed = [key hyp_parseRelationship];
         if (parsed.toMany) {
             if (type == HYPJSONNestedAttributesType) {
+                HYPParsedRelationship *parsedRelationshipPrefix = [key hyp_parseRelationship];
```

Comment:
Would it be possible to implement `compare:` on `HYPParsedRelationship`. It could be useful at times, just a thought.